//
//  CapstoneUnitsTests.swift
//  CapstoneUnitsTests
//
//  Created by 정성윤 on 2024/04/16.
//

import XCTest
//import RxSwift
//import RxCocoa
//@testable import Capstone

final class CapstoneUnitsTests: XCTestCase {
    //메인 테스트
//    var mainViewModel : MainViewModel!
//    var disposeBag : DisposeBag!
    //테스트 시작
    override func setUpWithError() throws {
//        mainViewModel = MainViewModel()
//        disposeBag = DisposeBag()
    }
    //테스트 종료
    override func tearDownWithError() throws {
//       mainViewModel = nil
//       disposeBag = nil
    }
    //테스트 성공
//    func testFeelingTriggerSuccess() throws {
//        let expectation = XCTestExpectation(description: "Feeling result is received")
//
//        mainViewModel.feelingResult
//            .subscribe(onNext: { result in
//                XCTAssertNotNil(result, "Expected non-nil result")
//                expectation.fulfill()
//            }, onError: { error in
//                XCTFail("Expected no error, but got \(error)")
//            })
//            .disposed(by: disposeBag)
//
//        mainViewModel.feelingTrigger.onNext(())
//
//        wait(for: [expectation], timeout: 5.0)
//    }
//    //테스트 실패
//    func testFeelingTriggerError() throws {
//        let expectation = XCTestExpectation(description: "Error is received")
//
//        mainViewModel.feelingResult
//            .subscribe(onNext: { result in
//                XCTFail("Expected error, but got result \(result)")
//            }, onError: { error in
//                expectation.fulfill()
//            })
//            .disposed(by: disposeBag)
//
//        mainViewModel.feelingTrigger.onNext(())
//
//        wait(for: [expectation], timeout: 5.0)
//    }
    
    func testExample() throws {
        
    }
    
    func testPerformanceExample() throws {
        measure {
            
        }
    }
    
}
